### PR TITLE
fix(web-next): use UK date locale and display format

### PR DESCRIPTION
**Pull request by iguliaev**
## Why

The app was still showing a Sunday-start calendar and US-style date strings in several places, which made date entry and reading inconsistent.

## What Changed

- Files or areas updated: `src/index.tsx`, app shell locale setup, shared date formatting utility, transaction/budget forms, list and show pages, header search results
- New functionality added: global en-GB locale support and a shared `DD/MM/YYYY` display formatter
- Existing behavior changed: date pickers now start on Monday and displayed dates render as `DD/MM/YYYY`

## Key Decisions

- Decision: apply the locale globally rather than per date picker
- Reasoning: it keeps the calendar behavior consistent across the app and avoids one-off overrides
- Alternatives considered: patching only the transaction page, but that would leave other date pickers inconsistent

## How This Affects the System

- User-facing changes: calendars start on Monday and date text shows in UK format
- API changes: none
- Performance implications: none
- Database changes: none
- Breaking changes: none

## Testing

- New tests added: none
- Existing tests status: TypeScript check passed; targeted ESLint on changed files passed with one pre-existing warning in the color-mode context file
- Manual testing performed: reviewed the updated date surfaces and locale wiring
- Edge cases tested: stored date strings remain `YYYY-MM-DD`; only presentation changed
- Test files modified or added: none

### CODE REVIEW

**Comment by iguliaev** [comment]
Review findings: the date-picker comments were valid — the new DD/MM/YYYY display needed to keep accepting ISO input used by the E2E helpers, so I updated the pickers to accept both formats and aligned the test helper/display expectations. I also checked CI; there isn't a failing run here, only a cancelled preview E2E job.

File: apps/web-next/src/pages/transactions/list.tsx [hunk 1]
```diff
@@ -179,6 +179,7 @@ export const TransactionList = () => {
               return (
                 <div style={{ padding: 8 }}>
                   <DatePicker.RangePicker
+                    format="DD/MM/YYYY"
                     value={value}
```
[the remark this answers is not in the document]
Comment (iguliaev) [reply apps/web-next/src/pages/transactions/list.tsx:183]: Fixed: the range picker now accepts both DD/MM/YYYY and YYYY-MM-DD, so the filter keeps the UK display while still parsing ISO input.

File: apps/web-next/src/pages/budgets/edit.tsx [hunk 2]
```diff
@@ -174,7 +174,7 @@ export const BudgetEdit = () => {
             value: value ? dayjs(value) : undefined,
           })}
         >
-          <DatePicker style={{ width: "100%" }} />
+          <DatePicker format="DD/MM/YYYY" style={{ width: "100%" }} />
         </Form.Item>
```
[the remark this answers is not in the document]
Comment (iguliaev) [reply apps/web-next/src/pages/budgets/edit.tsx:179]: Fixed: the start date field now uses the shared date formats, so it keeps the UK display and still parses ISO input.

File: apps/web-next/src/components/header/index.tsx [hunk 3]
```diff
@@ -91,13 +92,7 @@ const renderTransactionItem = (
...
   const note = t.notes ?? `Transaction #${t.id}`;
-  const formattedDate = t.date
-    ? new Date(t.date).toLocaleDateString("en-GB", {
-        day: "numeric",
-        month: "short",
-        year: "numeric",
-      })
-    : null;
+  const formattedDate = formatDisplayDate(t.date);
   const meta = [t.category_name, formattedDate].filter(Boolean).join(" · ");
```
[the remark this answers is not in the document]
Comment (iguliaev) [reply apps/web-next/src/components/header/index.tsx]: Fixed: the header now skips missing dates instead of rendering the placeholder in the search meta.

File: apps/web-next/src/pages/budgets/create.tsx [hunk 4]
```diff
@@ -135,10 +135,10 @@ export const BudgetCreate = () => {
           <InputNumber min={0.01} precision={2} style={{ width: "100%" }} />
         </Form.Item>
         <Form.Item label="Start Date" name="start_date">
-          <DatePicker style={{ width: "100%" }} />
+          <DatePicker format="DD/MM/YYYY" style={{ width: "100%" }} />
         </Form.Item>
         <Form.Item label="End Date" name="end_date">
-          <DatePicker style={{ width: "100%" }} />
+          <DatePicker format="DD/MM/YYYY" style={{ width: "100%" }} />
```
[the remark this answers is not in the document]
Comment (iguliaev) [reply apps/web-next/src/pages/budgets/create.tsx]: Fixed: budget date fields now serialize to YYYY-MM-DD from the form item, so stored values stay stable and timezone-safe.

File: apps/web-next/src/pages/budgets/create.tsx [hunk 5]
```diff
@@ -135,10 +135,10 @@ export const BudgetCreate = () => {
...
         </Form.Item>
```
[the remark this answers is not in the document]
Comment (iguliaev) [reply apps/web-next/src/pages/budgets/create.tsx]: Fixed: budget date pickers now accept both DD/MM/YYYY and YYYY-MM-DD, so the UI stays UK-formatted without breaking ISO paste or typing.

File: apps/web-next/src/utility/dateDisplay.ts [hunk 6]
```diff
@@ -0,0 +1,6 @@
+import dayjs from "dayjs";
+
+export const formatDisplayDate = (value?: string | null): string => {
+  if (!value) return "—";
+  return dayjs(value).format("DD/MM/YYYY");
+};
```
[the remark this answers is not in the document]
Comment (iguliaev) [reply apps/web-next/src/utility/dateDisplay.ts:7]: Fixed: invalid date strings now fall back to the placeholder instead of formatting to Invalid Date.

File: apps/web-next/src/pages/budgets/edit.tsx [hunk 7]
```diff
@@ -183,7 +183,7 @@ export const BudgetEdit = () => {
...
```
[the remark this answers is not in the document]
Comment (iguliaev) [reply apps/web-next/src/pages/budgets/edit.tsx:189]: Fixed: the end date field now uses the shared date formats, so it keeps the UK display and still parses ISO input.

File: apps/web-next/src/pages/transactions/create.tsx [hunk 8]
```diff
@@ -104,7 +104,7 @@ export const TransactionCreate = () => {
             value: value ? dayjs(value) : undefined,
           })}
         >
-          <DatePicker format="YYYY-MM-DD" />
+          <DatePicker format="DD/MM/YYYY" />
         </Form.Item>
```
[the remark this answers is not in the document]
Comment (iguliaev) [reply apps/web-next/src/pages/transactions/create.tsx:109]: Fixed: the date picker now accepts both DD/MM/YYYY and YYYY-MM-DD through a shared format list, so UK display stays intact while the existing ISO test input still parses.

File: apps/web-next/src/pages/transactions/edit.tsx [hunk 9]
```diff
@@ -130,7 +130,7 @@ export const TransactionEdit = () => {
             value: value ? dayjs(value) : undefined,
           })}
         >
-          <DatePicker format="YYYY-MM-DD" />
+          <DatePicker format="DD/MM/YYYY" />
         </Form.Item>
```
[the remark this answers is not in the document]
Comment (iguliaev) [reply apps/web-next/src/pages/transactions/edit.tsx:135]: Fixed: the edit form uses the same shared date formats, so DD/MM/YYYY display and ISO input both work here too.

File: apps/web-next/src/pages/budgets/edit.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (iguliaev) [reply apps/web-next/src/pages/budgets/edit.tsx:189]: Fixed: the edit form now serializes budget dates to YYYY-MM-DD from the form item, so unchanged and edited values behave consistently.

**Comment by iguliaev** [comment]
All review comments are addressed and replied to. The branch is pushed, the UK date display is kept, ISO input still works where needed, invalid dates now fall back to the placeholder, and budget dates serialize as YYYY-MM-DD. CI shows Vercel passing and the preview E2E job still running.
